evidence_test.go

```diff
@@ -146,67 +163,145 @@ func TestEvidence_sign_and_verify_realm_key_mismatch(t *testing.T) {
+	err := e.SetClaims(
+		mustBuildValidCcaPlatformClaims(t, true),
+		mustBuildValidCcaRealmClaims(t),
+	)
 	require.NoError(t, err)
 
 	expected := &testImplementationID
 
-	actual := EvidenceIn.GetImplementationID()
+	actual := e.GetImplementationID()
 	assert.Equal(t, expected, actual)
 }
 
 func TestEvidence_GetRealmPubKey_ok(t *testing.T) {
-	var EvidenceIn Evidence
+	var e Evidence
 
-	err := EvidenceIn.SetCcaRealmClaims(mustBuildValidCcaRealmClaims(t))
+	err := e.SetClaims(
+		mustBuildValidCcaPlatformClaims(t, true),
+		mustBuildValidCcaRealmClaims(t),
+	)
 	require.NoError(t, err)
 
 	expected := &testRAKPubRaw
 
-	actual := EvidenceIn.GetRealmPublicKey()
+	actual := e.GetRealmPublicKey()
 	assert.Equal(t, expected, actual)
 }
 
-func TestEvidence_sign_missing_realm_claims(t *testing.T) {
-	var (
-		EvidenceIn Evidence
-		unused     cose.Signer
-	)
+func TestEvidence_SetClaims_missing_realm_claims(t *testing.T) {
+	var e Evidence
 
-	err := EvidenceIn.SetCcaPlatformClaims(
+	err := e.SetClaims(
 		mustBuildValidCcaPlatformClaims(t, true),
+		nil,
+	)
+	assert.EqualError(t, err, "nil claims supplied")
+}
+
+func TestEvidence_SetClaims_missing_platform_claims(t *testing.T) {
+	var e Evidence
+
+	err := e.SetClaims(
+		nil,
+		mustBuildValidCcaRealmClaims(t),
 	)
+	assert.EqualError(t, err, "nil claims supplied")
+}
+
+func TestEvidence_SetClaims_bind_failed(t *testing.T) {
+	emptyRealmClaims := &RealmClaims{}
+	emptyPlatformClaims := &psatoken.CcaPlatformClaims{}
+
+	expectedErr := "tokens binding failed: computing binder value: extracting RAK from the realm token: missing mandatory claim"
+
+	var e Evidence
+
+	err := e.SetClaims(emptyPlatformClaims, emptyRealmClaims)
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestEvidence_SetClaims_invalid_platform(t *testing.T) {
+	emptyPlatformClaims := &psatoken.CcaPlatformClaims{}
+
+	expectedErr := "validation of cca-platform-claims failed: validating profile: missing mandatory claim"
+
+	var e Evidence
+
+	err := e.SetClaims(
+		emptyPlatformClaims,
+		mustBuildValidCcaRealmClaims(t),
+	)
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestEvidence_SetClaims_invalid_realm(t *testing.T) {
+	incompleteRealmClaims := &RealmClaims{}
+
+	// just set the bare minimum to compute the binder
+	err := incompleteRealmClaims.SetPubKey(testRAKPubRaw)
+	require.NoError(t, err)
+
+	err = incompleteRealmClaims.SetPubKeyHashAlgID("sha-256")
 	require.NoError(t, err)
 
-	_, err = EvidenceIn.Sign(unused, unused)
-	assert.EqualError(t, err, "missing realm claims in evidence")
+	expectedErr := "validation of cca-realm-claims failed: validating cca-realm-challenge claim: missing mandatory claim"
+
+	var e Evidence
+
+	err = e.SetClaims(
+		mustBuildValidCcaPlatformClaims(t, false),
+		incompleteRealmClaims,
+	)
+	assert.EqualError(t, err, expectedErr)
 }
 
-func TestEvidence_sign_missing_platform_claims(t *testing.T) {
+func TestEvidence_Sign_no_platform_claims(t *testing.T) {
```

Comment:
We should also add an equivalent here, with TestEvidence_no_realm_claims

Reply:
I wanted that too, but it's not possible to subvert the interface to get to that condition...